### ISSUE TITLE
' An I/O error occurred while sending to the backend.' on cascading delete for many nodes

### ISSUE THREAD

**Issue by Jaszkowic** [prisma/prisma#2953]
**Describe the bug**

Error when cascading delete is performed for a lot of related nodes.

> [36mprisma_1    |[0m org.postgresql.util.PSQLException: An I/O error occurred while sending to the backend.

> Caused by: java.io.IOException: Tried to send an out-of-range integer as a 2-byte value: 32768

**To Reproduce**
1. Setup Prisma 1.14.2 with latest PostgreSQL database.
2. Create schema:

```graphql
type Painting {
    id: String! @unique,
    strokes: [Stroke!]! @relation(name: "StrokesForPainting", onDelete: CASCADE)
}

type Stroke {
    id: String! @unique,
    painting: Painting! @relation(name: "StrokesForPainting"),
    samples: [StrokeSample!]! @relation(name: "StrokeSamplesForStroke", onDelete: CASCADE)
}

type StrokeSample {
    id: String! @unique,
    x: Float!,
    y: Float!,
    stroke: Stroke! @relation(name: "StrokeSamplesForStroke")
}
```
3. Seed with data. E.g. I had 128 strokes each with 256 samples in a painting.

4. Try to delete painting -> cascading deletes are performed. Everything is alright when number of strokes/samples are quite low (don't know exactly how many are okay). **Error occurred with 128 strokes each with 256 samples (in total 32768 samples).**

**Expected behavior**
Cascading delete should be performed successfully.

**Versions (please complete the following information):**
Prisma 1.14.2 with latest PostgreSQL database.

**Comment by marktani** [comment]
This looks like a regression of https://github.com/prisma/prisma/issues/2863.

**Comment by marktani** [comment]
I can reproduce this. This is not a regression of #2863, as this only happens for cascading deletes now.

You can reproduce this with the following datamodel:

```graphql
type Top {
  id: ID! @unique
  middles: [Middle!]! @relation(name: "TopMiddle", onDelete: CASCADE)
}

type Middle {
  id: ID! @unique
  top: Top! @relation(name: "TopMiddle")
  bottoms: [Bottom!]! @relation(name: "MiddleBottom", onDelete: CASCADE)
}

type Bottom {
  id: ID! @unique
  x: Float!
  y: Float!
  middle: Middle! @relation(name: "MiddleBottom")
}
```

Deploy it, then import the following zip by running `prisma import --data export.zip`:

[export.zip](https://github.com/prisma/prisma/files/2446418/export.zip)

Then run this mutation to reproduce the error:

```graphql
mutation a {
  deleteTop(where: {
    id: "cjmukxsgz00120a46zocky1js"
  }) {
    id
  }
}
```

This is the stacktrace:

<details><summary>Stacktrace</summary>

```
org.postgresql.util.PSQLException: An I/O error occurred while sending to the backend.
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:333)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:155)
        at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:144)
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
        at com.prisma.api.connector.jdbc.database.BuilderBase.$anonfun$deleteToDBIO$1(BuilderBase.scala:65)
        at com.prisma.api.connector.jdbc.database.BuilderBase.$anonfun$deleteToDBIO$1$adapted(BuilderBase.scala:65)
        at com.prisma.api.connector.jdbc.database.BuilderBase.$anonfun$jooqToDBIO$1(BuilderBase.scala:82)
        at slick.jdbc.SimpleJdbcAction.run(StreamingInvokerAction.scala:70)
        at slick.jdbc.SimpleJdbcAction.run(StreamingInvokerAction.scala:69)
        at slick.dbio.DBIOAction$$anon$4.$anonfun$run$3(DBIOAction.scala:239)
        at scala.collection.Iterator.foreach(Iterator.scala:929)
        at scala.collection.Iterator.foreach$(Iterator.scala:929)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
        at scala.collection.IterableLike.foreach(IterableLike.scala:71)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        at slick.dbio.DBIOAction$$anon$4.run(DBIOAction.scala:239)
        at slick.dbio.DBIOAction$$anon$4.run(DBIOAction.scala:237)
        at slick.basic.BasicBackend$DatabaseDef$$anon$2.liftedTree1$1(BasicBackend.scala:275)
        at slick.basic.BasicBackend$DatabaseDef$$anon$2.run(BasicBackend.scala:275)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Tried to send an out-of-range integer as a 2-byte value: 65536
        at org.postgresql.core.PGStream.sendInteger2(PGStream.java:224)
        at org.postgresql.core.v3.QueryExecutorImpl.sendParse(QueryExecutorImpl.java:1440)
        at org.postgresql.core.v3.QueryExecutorImpl.sendOneQuery(QueryExecutorImpl.java:1762)
        at org.postgresql.core.v3.QueryExecutorImpl.sendQuery(QueryExecutorImpl.java:1326)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:298)
        ... 25 more
```
</details>

**Comment by mavilein** [comment]
Thanks for reporting the issue and your thorough description! 🙏 This has been fixed in `1.18.1`.
